test/active_record_cache_find_by_test.rb

```diff
@@ -150,8 +155,9 @@ def assert_not_caches(key, cache = Rails.cache)
   end
 
   it('caches find_by_x with two attributes') do
+    skip 'was not cached on rails 4.2'
```

Comment:
not quite like this.

I put in some time last night to see if this would be an easy fix for 4.2. it's not. it's not worth fixing so I am removing this feature.